Combine CSV test diffing with new Pandas

## Bug Description
An update to pandas resulted in a failure in `test_combine_csv.TestCombineCSV` on mac (pandas 1.1.5 -> 1.2.0)

Using

```
pandas._testing.assert_frame_equal(df_test._final_df, gold_df)
```

instead for a fuzzy comparison leads to:

```
postprocessing/tests.combine_csv: AssertionError: DataFrame.columns are different
postprocessing/tests.combine_csv:
postprocessing/tests.combine_csv: DataFrame.columns values are different (33.33333 %)
postprocessing/tests.combine_csv: [left]:  Index(['4.397350132855e-05', '0.0002090060028648', '0.0004243273304685',
postprocessing/tests.combine_csv:        '0.0005893598320047', '0.0006773068346618', '0.0008423393361981',
postprocessing/tests.combine_csv:        '0.0010576606638019', '0.0012226931653381', '0.0013106401679952',
postprocessing/tests.combine_csv:        '0.0014756726695315', '0.0016909939971352', '0.0018560264986715'],
postprocessing/tests.combine_csv:       dtype='object')
postprocessing/tests.combine_csv: [right]: Index(['4.397350132855e-05', '0.0002090060028648', '0.00042432733046854',
postprocessing/tests.combine_csv:        '0.00058935983200478', '0.00067730683466188', '0.00084233933619813',
postprocessing/tests.combine_csv:        '0.0010576606638019', '0.0012226931653381', '0.0013106401679952',
postprocessing/tests.combine_csv:        '0.0014756726695315', '0.0016909939971352', '0.0018560264986715'],
postprocessing/tests.combine_csv:       dtype='object')
```

Four of the columns are diffing by ~1e-13, and because we're using columns to store actual data, the comparisons here are wonky.

## Steps to Reproduce
Run python tests on mac with current moose and updated conda (pandas 1.2.0).

## Impact
Testing failures.

--- a/python/postprocessing/tests/test_combine_csv.py
+++ b/python/postprocessing/tests/test_combine_csv.py
@@ -11,6 +11,7 @@
 import os
 import unittest
 import pandas
+import numpy
 try:
     from postprocessing import combine_csv
 except ModuleNotFoundError:
@@ -81,8 +82,21 @@ class TestCombineCSV(unittest.TestCase):
         self.assertTrue(df_test._ended)
         gold_df = pandas.read_csv("{}/combine_bilinear.csv".format(
             self.__goldpath))
-        self.assertTrue(df_test._final_df.equals(gold_df),
-            msg="Pandas dataframe is different from gold CSV for bilinear usage.")
+
+        self.assertTrue(df_test._final_df.shape == gold_df.shape,
+            msg="Pandas dataframe size is different from gold CSV for bilinear usage.")
+
+        # We used to use DataFrame.equals for a comparison here, but because a
+        # newer version of Pandas resulted in a very small diff in the column
+        # data, we can't use it anymore. Therefore - we need to do a more fuzzy
+        # check on both the columns and th data.
+        data_difference = df_test._final_df.to_numpy() - gold_df.to_numpy()
+        column_difference = df_test._final_df.columns.to_numpy(dtype=numpy.double) \
+                            - gold_df.columns.to_numpy(dtype=numpy.double)
+        self.assertTrue(abs(data_difference.max()) < 1e-10,
+            msg="Pandas data is different from gold CSV for bilinear usage.")
+        self.assertTrue(abs(column_difference.max()) < 1e-10,
+            msg="Pandas column data is different from gold CSV for bilinear usage.")
 
     def testBasenameError(self):
         """

